# Stefan Teleman <stefan.teleman@oracle.com>
# Port of CilkPlus (libcilkrts) to SPARC and Solaris.
# Internal patch. SPARC specific:
# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68945
--- libcilkrts/runtime/sysdep-unix.c	2013-10-30 15:51:29.000000000 -0700
+++ libcilkrts/runtime/sysdep-unix.c	2015-08-05 11:51:37.086142492 -0700
@@ -81,6 +81,10 @@
 #include <pthread.h>
 #include <unistd.h>
 
+#if defined(sun) || defined(__sun) || defined(__sun__)
+#include <strings.h>
+#endif
+
 #if defined HAVE_ALLOCA_H
 # include <alloca.h>
 #elif defined __GNUC__
@@ -110,6 +114,14 @@
 #   define MAP_ANONYMOUS MAP_ANON
 #endif
 
+#if defined(sun) || defined(__sun) || defined(__sun__)
+#include <sys/resource.h>
+#include <dlfcn.h>
+#ifndef MAP_ANONYMOUS
+#define MAP_ANONYMOUS MAP_ANON
+#endif
+#endif
+
 #ifdef  __VXWORKS__
 #   include <vxWorks.h>   
 #   include <vxCpuLib.h>  
